x/ibc/24-validation/validate.go

```diff
@@ -0,0 +1,61 @@
+package validation
+
+import (
+	"regexp"
+	"strings"
+)
+
+// regular expression to check string is lowercase alphabetic characters only
+var isAlphaLower = regexp.MustCompile(`^[a-z]+$`).MatchString
+
+// regular expression to check string is alphanumeric
+var isAlphaNumeric = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString
+
+// Validator function type to validate path and identifier bytestrings
+type Validator func(string) bool
```

Comment:
Should this function take in a byte array instead or string?

Should this function be returning an error instead of boolean?

Reply:
> Should this function take in a byte array instead or string?

it's ok to use a string.

> Should this function be returning an error instead of boolean?

I was thinking of replicating sort of like a  `ValidateBasic` logic that returns the exact error because this will mostly be used on `msg.ValidateBasic()`